captum/attr/_utils/gradient.py

```diff
@@ -175,9 +175,12 @@ def forward_hook(module, inp, out):
             # TODO we need to think what will be the best way of storing eval
             # tensors per device for each input per example. This implementation
             # doesn't support a tuple of inputs
-            saved_layer[eval_tsr.device] = eval_tsr
+            saved_layer[eval_tsr.device] = eval_tsr.clone()
 
-    hook = layer.register_forward_hook(forward_hook)
+    if attribute_to_layer_input:
+        hook = layer.register_forward_pre_hook(forward_hook)
+    else:
+        hook = layer.register_forward_hook(forward_hook)
     _run_forward(forward_fn, inputs, additional_forward_args=additional_forward_args)
```

Comment:
I was just looking into this. Doesn't `_run_forward ` also need `target_ind` to be passed as an argument ?